# Amazon Beauty Products Review-Sentiment Analysis

## Import Necessary Libraries

In [1]:
# Dataframe
import pandas as pd

# Array
import numpy as np

# Visualizations
import matplotlib.pyplot as plt
from matplotlib.colors import ListedColormap
import seaborn as sns
import matplotlib.colors as colors
%matplotlib inline

# Datetime
from datetime import datetime

## Warnings
import warnings
from scipy import stats
warnings.filterwarnings('ignore')

## Reading the Inspected Dataset 

In [2]:
# Read the file
df2 = pd.read_csv('C:/Users/guzel/Documents/amazon_beauty_products/data_inspected.csv')

In [3]:
# Sample observations
df2.head(3)

customer     product  rating  \
0   A6VPK7X53QNAQ  B0000CC64W     5.0   
1  A3CHMHGSJSQ02J  B0000CC64W     5.0   
2  A1V1EP514B5H7Y  B0000CC64W     5.0   

                                         review_text  pos_feedback  \
0  If I had to choose only one product to take ca...             5   
1  Makes my skin lovely and smooth As a woman nea...             2   
2  Works well at a reasonable price I've used thi...             0   

   neg_feedback rating_class        time  
0             0         good  2009-06-18  
1             0         good  2013-01-18  
2             0         good  2011-11-29

## Descriptive Statistics 

In [4]:
# Read statistic summary of numeric variables
df2.describe()

rating  pos_feedback  neg_feedback
count  28576.000000  28576.000000  28576.000000
mean       4.170493      0.948488      0.324888
std        0.992277      6.829651      1.105870
min        1.000000      0.000000      0.000000
25%        4.000000      0.000000      0.000000
50%        4.000000      0.000000      0.000000
75%        5.000000      1.000000      0.000000
max        5.000000    549.000000     52.000000

In [5]:
# Customer totals for each rating 
df2['rating'].value_counts()

5.0    13683
4.0     8742
3.0     4067
2.0     1508
1.0      576
Name: rating, dtype: int64

In [6]:
# Customer totals for each rating class
df2['rating_class'].value_counts()

good       22425
neutral     4067
bad         2084
Name: rating_class, dtype: int64

In [7]:
# Print the number of positive feedback
pos = len(df2[df2['pos_feedback'] > 0])
print('Number of positive feedbacks: {}'.format(pos))

# Print the number of negative feedback
neg = len(df2[df2['neg_feedback'] > 0])      
print('\nNumber of negative feedbacks: {}'.format(neg))

# Print the number of no 'helpful' feedback 
total = len(df2)
print('\nNumber of no helpful feedbacks: {}'.format(total - (pos + neg)))

Number of positive feedbacks: 8397

Number of negative feedbacks: 6078

Number of no helpful feedbacks: 14101


In [8]:
# Statistics of non-numeric variables

# Number of unique customers
print('\nNumber of unique customers : {}'.format(len(df2['customer'].unique())))
      
# Number of unique products
print('\nNumber of unique products : {}'.format(len(df2['product'].unique())))
      
# Review number per unique customer
print('\nReview per customer: {}'.format((len(df2)/len(df2['customer'].unique()))))      

# Review number per unique product 
print('\nReview per product: {}'.format((len(df2)/len(df2['product'].unique()))))     


Number of unique customers : 1340

Number of unique products : 733

Review per customer: 21.32537313432836

Review per product: 38.9849931787176


**Descriptive Statistic Summary:**

**- Rating Status:**

28576 customer gives ratings and mean of the ratings is 4.17, which means that customers prefer to give high ratings for products. Standard deviation and percentiles also show that 1 and 2 ratings for products are rare. To be able to predict the ratings reasonably, we classified them as 'good', 'neutral', and 'bad' above. 

According to the statistics on rating stars:

576 customers give 1 star 

1508 customers give 2 stars

4067 customers give 3 stars

8742  customers give 4 stars

13683 customers give 5 stars


On the other hand, if we look for the rating class, the picture is look like below.

2084 customers give bad ratings

4067 customers give neutral ratings

22425 customers give good ratings

**- Feedback Status**

8397 customers find the given reviews logical and totally agree with them. They give positive feedbacks. The mean 0.9 and standard deviation is 6.8 whereas the highest positive feedback number for reviews is 549. That shows we may have some outliers in the dataset. 

6078 customers find the given reviews useless and don't aggree with the them. They give negative feedbacks. The mean 0.3 and standard deviation is 1.1 whereas the highest negative feedback number for reviews is 52.

On the other hand 14101 customers don't give either positive or negative feedbacks for reviews. 

**- Non-numeric variables statistics:**

We have 1340 unique customers and 733 products in this dataset. Each customers averagely give 21 reviews for products and on the other hand, there is averagely 40 reviews for each product in the website. 

On exploratory data analysis, we will check if there is a case for outliers by visualizing data. 